ecommerce-ui/src/App.js

```diff
@@ -0,0 +1,54 @@
+import React, { Component } from 'react';
+// import logo from './logo.svg';
+import './App.css';
+import Room from './Room.js';
+import airBnbs from './airbnbs.json';
+import Cart from './Cart.js';
+
+class App extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.addRoom = this.addRoom.bind(this);
+
+    this.state = {
+      rooms: []
+    };
+  }
+
+  addRoom = (room) => {
```

Comment:
The easiest way to deal with binding functions is to use the arrow function syntax you used here only. By doing so, you shouldn't need to use `this.addRoom = this.addRoom.bind(this)`, although the function will still behave the same because you're binding to the same context.